doc/source/data/batch_inference.rst

```diff
@@ -1,823 +1,455 @@
-    ``"pandas"`` batch format, you will need to know the basics of interacting with
-    dataframes to make your batch inference jobs work.
-
-.. seealso::
-
-    As we've discussed in this guide, using :meth:`ds.map_batches() <ray.data.Dataset.map_batches>`
-    on a class defining your model
-    should be your default choice for running inference with Ray.
-    For instance, if you're already using the Ray AIR framework for running your ML workflows,
-    you may want to use the
-    :ref:`framework-specific batch predictor implementations<air_framework_predictors>`.
-
-    To see an extension of the quick start example using an AIR
-    ``HuggingFacePredictor``, see the following example:
-
-    .. dropdown:: Batch inference example with HuggingFace and Ray AIR
-
-        .. literalinclude:: ./doc_code/hf_quick_start.py
-            :language: python
-            :start-after: __hf_quickstart_air_start__
-            :end-before: __hf_quickstart_air_end__
-
-.. _batch_inference_config:
-
-Configuration & Troubleshooting
--------------------------------
+.. _batch_inference_batch_size:
 
 Configuring Batch Size
 ~~~~~~~~~~~~~~~~~~~~~~
 
-An important parameter to set for :meth:`ds.map_batches() <ray.data.Dataset.map_batches>`
-is ``batch_size``, which controls the size of the batches provided to the function.
-Here's a simple example of loading the IRIS dataset (which has Pandas format by default)
-and processing it with a batch size of `10`:
-
-.. literalinclude:: ./doc_code/batch_formats.py
-  :language: python
-  :start-after: __simple_map_function_start__
-  :end-before: __simple_map_function_end__
-
-Increasing ``batch_size`` can result in faster execution by better leveraging vectorized
-operations and hardware, reducing batch slicing and concatenation overhead, and overall
-saturation of CPUs or GPUs.
-On the other hand, this will also result in higher memory utilization, which can
-lead to out-of-memory (OOM) failures.
-If encountering OOMs, decreasing your ``batch_size`` may help.
-
-.. caution::
-  The default ``batch_size`` of ``4096`` may be too large for datasets with large rows
-  (e.g. tables with many columns or a collection of large images).
-
-
-.. _batch_inference_gpu:
-
-Using GPUs in batch inference
------------------------------
-
-To use GPUs for inference, first pdate the callable class implementation to
-move the model and data to and from Cuda device.
-Here's a quick example for a PyTorch model:
+Configure the size of the input batch that is passed to ``__call__`` by setting the ``batch_size`` argument for :meth:`ds.map_batches() <ray.data.Dataset.map_batches>`
 
-.. code-block:: diff
+Increasing batch size results in faster execution because inference is a vectorized operation. For GPU inference, increasing batch size increases GPU utilization. Set the batch size to as large possible without running out of memory. If you encounter OOMs, decreasing the ``batch_size`` may help.
```

Comment:
```suggestion
Increasing batch size results in faster execution because inference is a vectorized operation. For GPU inference, increasing batch size increases GPU utilization. Set the batch size to as large possible without running out of memory. If you encounter OOMs, decreasing ``batch_size`` may help.
```